hera_qm/xrfi.py

```diff
@@ -421,10 +451,104 @@ def detrend_meanfilt(data, flags=None, Kt=8, Kf=8):
+    effects that would be very drastic if one were to call detrend_medfilt with
+    large kernel size.
+
+    Parameters
+    ----------
+    data : array
+        2D data array to process.
+    flags : array, optional
+        2D flag array to be interpretted as mask for d. ONLY used for the regular
+        zscore (not modified).
+    modified : bool, optional
+        Whether to calculate the modified z-scores. Default is False.
+
+    Returns
+    -------
+    out : array
+        An array containing the outlier significance metric. Same type and size as d.
+
+    """
+    data = np.array(data)  # makes a copy of the data
+    if flags is not None:
+        data[flags] = np.nan
+    if modified:
+        if np.any(np.iscomplex(data)):
+            med_r = np.nanmedian(data).real
+            med_i = np.nanmedian(data).imag
+            mad_r = np.nanmedian(np.abs(data.real - med_r))
+            mad_i = np.nanmedian(np.abs(data.imag - med_i))
+            mad = np.sqrt(mad_r**2 + mad_i**2)
+            d_rs = data - med_r - 1j * med_i
+        else:
+            med = np.nanmedian(data)
+            mad = np.nanmedian(np.abs(data - med))
+            d_rs = data - med
+        # don't divide by zero, instead turn those entries into +inf
+        out = robust_divide(d_rs, np.array([1.486 * mad]))
+    else:
+        d_rs = data - np.nanmean(data)
+        out = robust_divide(d_rs, np.array([np.nanstd(data)]))
+    out[np.isnan(out)] = np.inf  # turn all nans into infs
+    return out
+
+
+def modzscore_1d(data, flags=None, kern=8, detrend=True):
+    """Calculate modified zscores in 1d.
+
+    Parameters
+    ----------
+    data : array
+        1D data array to detrend.
+    flags : array, optional
+        1D flag array to be interpretted as mask for d. NOT USED in this function,
+        but kept for symmetry with other preprocessing functions.
+    kern : int, optional
+        The box size to apply medfilt over. Default is 8 pixels.
+    detrend : bool, optional
+        Whether to detrend the data before calculating zscores. Default is True.
+        Setting to False is equivalent to an infinite kernel, but the function
+        does it more efficiently.
+
+    Returns
+    -------
+    zscore : array
+        An array containing the outlier significance metric. Same type and size as d.
+    """
+    if detrend:
+        # Delay import so scipy is not required for use of hera_qm
+        from scipy.signal import medfilt
+
+        kern = _check_convolve_dims(data, kern)
+        data = np.concatenate([data[kern - 1::-1], data, data[:-kern - 1:-1]])
+        # detrend in 1D. Do real/imag regardless of whether data are complex because it's cheap.
+        d_sm_r = medfilt(data.real, kernel_size=2 * kern + 1)
+        d_sm_i = medfilt(data.imag, kernel_size=2 * kern + 1)
+        d_sm = d_sm_r + 1j * d_sm_i
+        d_rs = data - d_sm
+        d_sq = np.abs(d_rs)**2
+        sig = np.sqrt(medfilt(d_sq, kernel_size=2 * kern + 1) / .456)
```

Comment:
I know it's common in the codebase, but I think it'd be good to add a reference in the docstring or explanatory comment inline for dividing by 0.456 (b/c it's a modified z-score), which otherwise looks like a magic number.